### ISSUE TITLE
.ini/.conf files not working

### ISSUE THREAD

**Issue by n8ur** [JensReimann/RTKLIB#15]

**Comment by n8ur** [comment]
Both the debian version (2.4.3b8) and current version of rtklib apps seem not to handle config files properly.  Using the -i option does not cause them to be read at startup, and it appears that sometimes trying to read the file actually ends up rewriting it with a whole bunch of additional lines that are not parseable and cause errors on the next startup.

I'm going to dig into this and will post a fix if/when I figure one out.

**Comment by n8ur** [comment]
I've learned what's going on; probably not a bug per se but cross-platform weirdness.  The QSettings object that reads/writes the .ini (or .config) file expects the file to have sections.  The Windows-based conf file doesn't.  When you specify the file at runtime with the -i switch, the file is read but not properly processed.  The cure is to create a new file from scratch by simply running rtk_navi (and I presume the other GUI programs).  If you make any changes while running the program, they will automagically be saved as rtknavi_qt.ini in the directory where the executable resides.  There's really little reason to hand edit the file.

There are three things that I think should be fixed, and I will submit a patch for them:
1.  Change the default extension from ".ini" to ".conf" to be consistent with *nix practice.
2.  Move the conf file storage location to somewhere like $HOME/.rtknavi_qt.conf -- having it try to write to /usr/bin/ is a bad idea.
3.  The file dialog that opens when you try to save the conf file within the program doesn't let you create a new file; it can only write to an existing one.

**Comment by JensReimann** [comment]
Hi n8ur,
thanks for you comments. Although the config options are named the same as in original rtklib, it was never intended to use the same config files for both, qt and non-qt, apps. I preferred the clean code base possible with QSettings.

One can replace the .ini file path generation 
    IniFile = fi.absolutePath() + "/" + fi.baseName() + ".ini";
in the constructor, with
    QSettings tempSettings(QSettings::IniFormat, QSettings::UserScope, "rtknavi-qt", "rtklib");
    IniFile = tempSettings.fileName(); 
to overcome 2.). Still the file is named .ini. If one replaces QSettings::IniFormat -> QSettings::NativeFormat it becomes .conf on Unix, but on Windows it would result in QSettings writing to registry instead of a .ini file. I'd like to avoid this.
Issue 3.) is already fixed in the latest commit.

**Comment by n8ur** [comment]
Hi Jens --

Thanks for the reply!  I have been working on making new Debian packages porting your latest code into the current 2.4.3b8 version.  When that's finished, I'll send you links to the source and packages (I'm also sending them to the Debian maintainer).  

To deal with the config file issues, I added a "LINUX" define to the makefiles and used that to change the behaviour.  On Windows, .ini files are created just as before.  On Linux, the config files are in $HOME/.<filename>.conf.  I get the executable filename from the OS, so if you want to run multiple instances calling them e.g. rtknavi_qt-1 and rtknavi_qt-2 will create and use separate conf files when each is run.

I had to code it slightly differently for command line than qt programs because the command line versions don't have the QT libs.  I've attached the code fragments that I pasted into each program type here.  (I'm no great programmer, but they seem to work).

Thanks much for your work on this!  I got into this project simply because I needed a newer version than what was available in Debian for some work I am doing, and I wanted to have the programs packaged for easy installation on multiple machines (and also hopefully a Raspberry Pi version).

[confdir.txt](https://github.com/JensReimann/RTKLIB/files/3125123/confdir.txt)
[confdir_qt.txt](https://github.com/JensReimann/RTKLIB/files/3125124/confdir_qt.txt)
